src/SAML2/Assertion.php

```diff
@@ -1462,6 +1493,16 @@ private function addAttributeStatement(DOMElement $root)
                 $attribute->setAttribute('NameFormat', $this->nameFormat);
             }
 
+            if ($name === SAML2_Const::EPTI_URN_MACE || $name === SAML2_Const::EPTI_URN_OID) {
+                foreach ($values as $eptiValue) {
+                    $attributeValue = $document->createElementNS(SAML2_Const::NS_SAML, 'saml:AttributeValue');
+                    $attribute->appendChild($attributeValue);
+                    SAML2_Utils::addNameId($attributeValue, $eptiValue);
```

Comment:
This merge had, unfortunately, two side effects:

* The first one is that this line is expecting you to encode the contents of an `eduPersonTargetedID` **always** as an array with some specific indexes. This breaks backwards-compatibility, because if you are creating an assertion manually (instead of parsing it from the original XML) and call `toXML()`, it will subsequently call `addAttributeStatement()`, which will then expect you to encode the attribute's value in a very specific way. Even worse, if the attribute's value is not an array, it will break completely with a white screen of death, since the second parameter of `SAML2_Utils::addNameId()` is declared as an array.

* On the other hand, this was tagged as `v1.10`, meaning everyone updating their installation via composer with a version requirement of `~1.0` or `^1.0` will install this new version, which could break their deployment. SimpleSAMLphp is an example of this, of course, and I've just received reports that people installing SSP 1.14.8 as an IdP and updating with composer are having trouble when trying to send an `eduPersonTargetedID` attribute.

We need to address this, in this branch and in 2.0. Master has it solved currently [thanks to the last commits](https://github.com/simplesamlphp/saml2/blob/master/src/SAML2/Assertion.php#L1513), since now an object is used and therefore `addNameId()` and `parseNameId()` are no longer necessary, but the code needs to avoid making any assumptions, since the attribute value can be provided by external parties.

Once this is fixed, we need a new release (v1.10.1? v1.11?) so that people can update and they can get their code working again (specially critical for SimpleSAMLphp acting as an IdP).